### ISSUE TITLE
Version 3.0.2 generates error when using default attributes

### ISSUE THREAD

**Issue by czjim** [serverdensity/chef-serverdensity#27]
Hi,

With version 3.0.2 the following error occurs in the /var/log/sd-agent/collector.log on CentOS 6.x. This error is generated with default attributes set.

```
2016-09-22 11:48:07 CEST | ERROR | sd.collector | config(config.py:859) | Unable to parse yaml config in /etc/sd-agent/conf.d/directory.yaml
Traceback (most recent call last):
  File "/usr/share/python/sd-agent/config.py", line 857, in _load_file_config
    check_config = check_yaml(config_path)
  File "/usr/share/python/sd-agent/util.py", line 170, in check_yaml
    raise Exception('You need to have at least one instance defined in the YAML file for this check')
Exception: You need to have at least one instance defined in the YAML file for this check
```

The process will continue to run however. We have updated from version 3.0.1 to 3.0.2. The changes from a chef-client run, see below:

```
Recipe: serverdensity::default
  * chef_gem[rest-client] action install (up to date)
  * yum_repository[serverdensity] action create
    * template[/etc/yum.repos.d/serverdensity.repo] action create (up to date)
    * execute[yum clean metadata serverdensity] action nothing (skipped due to action :nothing)
    * execute[yum-makecache-serverdensity] action nothing (skipped due to action :nothing)
    * ruby_block[yum-cache-reload-serverdensity] action nothing (skipped due to action :nothing)
     (up to date)
  * yum_package[sd-agent] action install (up to date)
  * service[sd-agent] action nothing (skipped due to action :nothing)
  * serverdensity[hostname.org] action update
    - fetch Server Density agent key
  Recipe: <Dynamically Defined Resource>
    * template[/etc/sd-agent/conf.d/agent.cfg] action create
      - update content in file /etc/sd-agent/conf.d/agent.cfg from 28081e to c456bf
      --- /etc/sd-agent/conf.d/agent.cfg	2016-09-08 10:16:49.865005466 +0200
      +++ /tmp/chef-rendered-template20160922-22057-17vfarj	2016-09-22 11:47:56.120873757 +0200
      @@ -25,5 +25,5 @@
       # if syslog is enabled but a host and port are not set, a local domain socket
       # connection will be attempted
       log_to_syslog: yes
      -listen_port: 17123
      +listen_port: 17124
      - restore selinux security context
    * link[/etc/sd-agent/config.cfg] action create (up to date)
    * service[sd-agent] action restart
      - restart service service[sd-agent]
  
Recipe: serverdensity::default
  * yum_package[sd-agent-directory] action install (up to date)
  * template[/etc/sd-agent/conf.d/directory.yaml] action create
    - update content in file /etc/sd-agent/conf.d/directory.yaml from 63fa92 to 0673ef
    --- /etc/sd-agent/conf.d/directory.yaml	2016-09-08 10:17:00.005131488 +0200
    +++ /tmp/chef-rendered-template20160922-22057-x5spmd	2016-09-22 11:48:04.482976762 +0200
    @@ -17,9 +17,4 @@
       # "filegauges" - boolean, when true stats will be an individual gauge per file (max. 20 files!) and not a histogram of the whole directory. default False
       # "pattern" - string, the `fnmatch` pattern to use when reading the "directory"'s files. default "*"
       # "recursive" - boolean, when true the stats will recurse into directories. default False
    -  - directory: ""
    -    # name: "tag_value"
    -    # filegauges: False
    -    # pattern: "*.log"
    -    # recursive: True
    - restore selinux security context
```

**Comment by harryperks** [comment]
hey Jim we're working on a fix for this.

**Comment by shurl** [comment]
Hi Jim, 

I've fixed this and released v3.0.3 both on GitHub and the Chef Supermarket. 

Any other issues let us know!
